Zend/zend_inheritance.c

```diff
@@ -1157,15 +1163,20 @@ static zend_always_inline inheritance_status do_inheritance_check_on_method_ex(
 		} while (0);
 	}
 
-	/* Prevent derived classes from restricting access that was available in parent classes (except deriving from non-abstract ctors) */
-	if (!checked && check_visibility
-			&& (child_flags & ZEND_ACC_PPP_MASK) > (parent_flags & ZEND_ACC_PPP_MASK)) {
+	/* Prevent derived classes from restricting access that was available in parent classes
+	 * (except deriving from non-abstract ctors). */
+	uint32_t ppp_mask = ZEND_ACC_PPP_MASK;
+	if (!checked && check_visibility && (child_flags & ppp_mask) > (parent_flags & ppp_mask)) {
```

Comment:
Not sure if that's really an improvement.

Reply:
An earlier version used `ppp_mask` in more places, but eventually they were removed. In that context, it makes sense, but as things stand now, I agree, let's revert this part.